internal/app/parcel/repository_test.go

```diff
@@ -33,41 +34,61 @@ func TestRepository_InsertParcel(t *testing.T) {
 		defer db.Close()
 
 		sqlxDB := sqlx.NewDb(db, "sqlmock")
-		m.ExpectExec("INSERT INTO parcel (.+) VALUES (.+)").
-			WithArgs(parcel.UserID, parcel.SourceAddress, parcel.DestinationAddress, parcel.SourceTime, parcel.ParcelType, parcel.Price, parcel.CarrierFee, parcel.CompanyFee).
-			WillReturnResult(sqlmock.NewResult(1, 1))
+		m.ExpectPrepare("INSERT INTO parcel (.+) VALUES (.+) RETURNING .+").ExpectQuery().
+			WillReturnRows(sqlmock.NewRows([]string{"user_id",
+				"source_address",
+				"destination_address",
+				"source_time",
+				"type",
+				"price",
+				"carrier_fee",
+				"company_fee",
+				"created_at",
+				"updated_at"}).
+				AddRow(parcel.UserID,
+					parcel.SourceAddress,
+					parcel.DestinationAddress,
+					parcel.SourceTime,
+					parcel.ParcelType,
+					parcel.Price,
+					parcel.CarrierFee,
+					parcel.CompanyFee,
+					parcel.CreatedAt,
+					parcel.UpdatedAt))
 
 		repo := NewRepository(sqlxDB)
-		err := repo.InsertParcel(context.Background(), parcel)
+		result, err := repo.InsertParcel(context.Background(), parcel)
+
 		assert.Nil(t, err)
+		assert.Equal(t, result, parcel)
```

Comment:
Did you check difference between `Equal` and. `EqualValues`?

Reply:
Yes, equal  checks the type also. If type is same, should use equal.